Урок 5. Проверка статистических гипотез. P-значения. Доверительные интервалы. A/B-тестирование

1. Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16. Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95, если выборочная средняя M = 80, а объем выборки n = 256.

Формула для рассчета доверительного интервала (при условии, что мы знаем средне квадратичное отклонение) будет $$\overline{X} \pm Z_{\frac{a}{2}} \cdot \frac{\sigma}{\sqrt{n}}$$

То-есть для наших исходных данных будет: $80 \pm 1.96 \cdot \frac{16}{\sqrt{256}}$

Наш интервал получается [78.04; 81.96]

2. В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные: 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1 Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95.

Посчитаем выборочную среднюю:

In [28]:
selection = [6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1]
X_e = sum(selection)/len(selection)
print(X_e)

6.589999999999999


Так же надо посчитать дисперсию.. 

In [29]:
D_e = sum(pow(x - X_e ,2) for x in selection)/len(selection)
print(D_e)

0.1829000000000001


Нам потребуется несмещенная оценка дисперсии

In [15]:
S_2 = 10/9 * D_e
print(S_2)

0.20322222222222233


Исправленное среднее квадратическое отклонение будет:

In [19]:
from math import sqrt
s = sqrt(S_2)
print(s)

0.4508017549014448


Интервал будет рассчитываться по той же формуле: перепишем ее в таком виде $$\overline{X_e} - \frac{t \cdot s}{\sqrt{n}} < a < \overline{X_e} + \frac{t \cdot s}{\sqrt{n}}$$

Коэффициент t возьмем из таблицы значений распределения Стьюдента: для 0.95 доверительной вероятности и выборке 10 - t будет 2.262

Подставим все и вычислим:

In [22]:
X_e - (2.262 * s)/sqrt(10)

6.267538255912425

In [23]:
X_e + (2.262 * s)/sqrt(10)

6.912461744087573

Получается истинная величина X лежит в пределах (6.267538255912425, 6.912461744087573) с доверительной вероятностью 0.95

3. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 кв.мм.

Тут практически задача с вебинара )

Итак у нас две гипотезы: H0: $\mu = \mu_0$ и H1: $\mu > \mu_0$ поскольку односторонний критерий

Выбираем статистический критерий: Раз нам известна дисперсия то критерий будет Z - тест

Определяем Z табличное: оно будет равно 1.645

Определяем рассчетное значение Z по формуле: $$Z_H = \frac{\overline{X} - \mu_0}{\frac{\sigma}{\sqrt{n}}}$$

В с нашими данными получается: 

In [25]:
((17.5 - 17) * sqrt(100))/sqrt(4)

2.5

Z наблюдаемое лежит гораздо правее Z табличного - значит верна гипотеза H1

4. Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет: 202, 203, 199, 197, 195, 201, 200, 204, 194, 190. Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?

Итак гипотеза H0 - продавец прав ) 

Гипотеза H1 - продавец не прав $\mu \neq \mu_0$

Поскольку дисперсии генеральной совокупности у нас нет - то критерий будет T (Стьюдента)

Тогда получаем T табличное: при доверительной вероятности и выборке 10 пачек - T табличное будет равно: 3.250 (с учетом несмещения)

Тогда считаем T наблюдаемое по формуле: $$\overline{X} \pm t_{\frac{a}{2}} \cdot \frac{\sigma}{\sqrt{n}}$$

Считаем среднее выборочное:

In [30]:
cookies = [202, 203, 199, 197, 195, 201, 200, 204, 194, 190]

In [32]:
XX_e = sum(cookies)/len(cookies)
print(XX_e)

198.5


Считаем дисперсию выборки:

In [33]:
DD_e = sum(pow(x - XX_e ,2) for x in cookies)/len(cookies)
print(DD_e)

17.85


In [37]:
# Смещаем ее
DD_e *= 9/10

Получаем интервал:

In [38]:
XX_e + 3.250 * (DD_e/sqrt(10))

215.01064694844663

In [39]:
XX_e - 3.250 * (DD_e/sqrt(10))

181.98935305155337

Вот дальше некая непонятность.. логически размышляя - раз мат ожидание генеральной совокупности (200) попадает в наш доверительный интервал при заданной доверительной вероятности - то гипотеза H0 верна - тоесть продавец не соврал )

Но возможно как и в случае с применением Z критерия - нужно было найти каким-то образом T наблюдаемое - и сравнить с T табличным... вот этого я не совсем понял как сделать.

После разбора ДЗ: а также точно как и с Z критерием ) $$t_H = \frac{\overline{X} - \mu_0}{\frac{\sigma}{\sqrt{n}}}$$
